crates/sui-adapter/src/adapter.rs

```diff
@@ -677,12 +661,7 @@ pub fn resolve_and_type_check(
     let fdef = match fdef_opt {
         Some(fdef) => fdef,
         None => {
-            return Err(SuiError::FunctionNotFound {
-                error: format!(
-                    "Could not resolve function '{}' in module {}",
-                    function, &module_id,
-                ),
-            })
+            return Err(ExecutionErrorKind::FunctionNotFound.into());
```

Comment:
My worry for losing the strings in this case, is that we really don't have another way to help people debug this. The only way devs can check their transaction is really by calling this function (which could be called from outside of execution, which I think does happen somewhere)

I think getting messages back here is probably more important for this than for the verifier

Reply:
Is there a reason this `ExecutionError::new_with_source` won't work here?

Reply:
no there isn't a reason, other than when i started this PR i went one route and then backtracked so i could indeed change this to preserve this information.

Reply:
Should be fixed now